pkgs/by-name/xm/xmlstarlet/package.nix

```
@@ -33,6 +33,11 @@ stdenv.mkDerivation rec {
       url = "https://sourceforge.net/p/xmlstar/patches/_discuss/thread/890e29655a/66ca/attachment/0001-Fix-build-with-libxml2-2.12.patch";
       hash = "sha256-XEk7aFOdrzdec1j2ffERJQbLH0AUNJA52QwA9jf4XWA=";
     })
+    (fetchurl {
```

```suggestion
    (fetchpatch {
```

does this actually *need* to be `fetchurl`? i don't quite understand yet when to use fetchpatch  or fetchurl. Might well be correct, but needs asking.

---

A better question to ask is it if needs to be `fetchpatch`.  `fetchurl` is simpler and has fewer dependencies, so when we know the patch is static it's better to use it. (This is explained in pkgs/README.md btw.)

---

shouldn't all patches be static, considering we give them FOD hashes?
Also, that very readme says:
> patches already merged upstream or published elsewhere should be retrieved using fetchpatch2

no mention about fetchurl as a  recommended practice....

---

I see your reasoning, and i accept it. I suppose this is another instance of https://github.com/NixOS/nixpkgs/issues/387072.

---

Yes, typically the only reason to use `fetchpatch*` is to avoid the hash changing from time to time, as it happens when fetching commits from GitHub, GitLab, etc.  (Sometimes you can also use other `fetchpatch*` features that modify the patch.)